src/modules/cmdpal/Microsoft.CmdPal.UI/IIconCacheService.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CmdPal.Extensions;
+using Microsoft.UI.Xaml.Controls;
+
+// To learn more about WinUI, the WinUI project structure,
+// and more about our project templates, see: http://aka.ms/winui-project-info.
+namespace Microsoft.CmdPal.UI;
+
+public interface IIconCacheService
+{
+    public Task<IconSource?> GetIconSource(IconDataType icon);
+}
```

Comment:
Do we need an interface here? Do we expect this to by used by the VMs or only the View layer?

I guess this is a question about the future intent of this service. As if the only thing that's going to use this is the Loading Behavior, it can own its lifetime too and not need to be part of the service registry.

Reply:
Ah, you know I guess that makes sense. I think I originally _felt_ like it was just gonna need to be scattered around more, but that Behavior is so tidy that we didn't need the service after all